web/src/app/alerts/AlertsList.js

```diff
@@ -0,0 +1,369 @@
+import React, { useState } from 'react'
+import { PropTypes as p } from 'prop-types'
+import gql from 'graphql-tag'
+import {
+  Hidden,
+  Snackbar,
+  SnackbarContent,
+  Typography,
+  makeStyles,
+  isWidthDown,
+} from '@material-ui/core'
+import {
+  ArrowUpward as EscalateIcon,
+  Check as AcknowledgeIcon,
+  Close as CloseIcon,
+  Info as InfoIcon,
+} from '@material-ui/icons'
+import { useSelector } from 'react-redux'
+
+import AlertsListFilter from './components/AlertsListFilter'
+import AlertsListControls from './components/AlertsListControls'
+import CreateAlertFab from './CreateAlertFab'
+import UpdateAlertsSnackbar from './components/UpdateAlertsSnackbar'
+
+import { formatTimeSince } from '../util/timeFormat'
+import { urlParamSelector } from '../selectors'
+import { useMutation, useQuery } from '@apollo/react-hooks'
+import QueryList from '../lists/QueryList'
+import useWidth from '../util/useWidth'
+
+export const alertsListQuery = gql`
+  query alertsList($input: AlertSearchOptions) {
+    alerts(input: $input) {
+      nodes {
+        id
+        alertID
+        status
+        summary
+        details
+        createdAt
+        serviceID
+        service {
+          name
+        }
+      }
+
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+    }
+  }
+`
+
+const updateMutation = gql`
+  mutation UpdateAlertsMutation($input: UpdateAlertsInput!) {
+    updateAlerts(input: $input) {
+      alertID
+      id
+    }
+  }
+`
+
+const escalateMutation = gql`
+  mutation EscalateAlertsMutation($input: [Int!]) {
+    escalateAlerts(input: $input) {
+      alertID
+      id
+    }
+  }
+`
+
+const useStyles = makeStyles(theme => ({
+  snackbar: {
+    backgroundColor: theme.palette.primary['500'],
+    height: '6.75em',
+    width: '20em', // only triggers on desktop, 100% on mobile devices
+  },
+  snackbarIcon: {
+    fontSize: 20,
+    opacity: 0.9,
+    marginRight: theme.spacing(1),
+  },
+  snackbarMessage: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+}))
+
+function getStatusFilter(s) {
+  switch (s) {
+    case 'acknowledged':
+      return ['StatusAcknowledged']
+    case 'unacknowledged':
+      return ['StatusUnacknowledged']
+    case 'closed':
+      return ['StatusClosed']
+    case 'all':
+      return ['StatusAcknowledged', 'StatusUnacknowledged', 'StatusClosed']
+    // active is the default tab
+    default:
+      return ['StatusAcknowledged', 'StatusUnacknowledged']
+  }
+}
+
+export default function AlertsList(props) {
+  const classes = useStyles()
+  const width = useWidth()
+  const isFullScreen = isWidthDown('md', width)
+
+  // state initialization
+  const [errorMessage, setErrorMessage] = useState('')
+  const [updateMessage, setUpdateMessage] = useState('')
+  const [updateComplete, setUpdateComplete] = useState(false)
```

Comment:
Instead of trying to track/keep in sync with these, let's switch to using the `useMutation` API/status/errors directly.

Reply:
We can do that, but I found it simpler to read as opposed to showing the update based off of three variables for the three mutations, as opposed to the one variable in state

Reply:
I think we should. With state, we are duplicating data that we already have from `useMutation`, and proper behavior depends on us keeping it in sync from inside the Promise callbacks.

As we are switching to hooks, we should be consistent in following the new API and removing usages of Promises with Apollo.

Reply:
So turns out we do need it in state as we require a manual closing of the snackbar– I can update the variable name so it's more clear that this is more than just duplicate data in state.